### ISSUE TITLE
[p0] Write a python training script which pickle.loads and pickle.dumps the gpt-2 weights, compatible with hugging face api GPT2LMHeadModel

### ISSUE THREAD

**Issue by annh3** [annh3/ai_planning_searching#21]
Make a plan for which third party packages you will use, i.e. how much detail you want to go into when writing this script.

According to [this](https://github.com/annh3/ai_planning_searching/issues/20) you want to tune adam optimizer's beta_1, beta_2, and epsilon 

Okay, here are some steps:

1. Read this: https://huggingface.co/transformers/v3.3.1/training.html#pytorch
2. Write a training_script sketch to get the high level format of the training script, without instantiating specific optimizers
3. Look into available code for adam optimizers and choose one where you can tune beta_1, beta_2, epsilon. The AdamW() optimizer provided by the hugging face library allows for tuning beta_1, beta_2, epsilon 💯 : https://huggingface.co/transformers/v3.3.1/main_classes/optimizer_schedules.html#transformers.AdamW.
4. Write a basic version of the script using the hugging face Trainer(), which handles epochs, learning rate scheduling, weight decay, logging directory, et cetera.
5. Test this training script on a simple text-to-text autogressive modeling task

**Comment by annh3** [comment]
Updated in https://github.com/annh3/ai_planning_searching/issues/27.
